benchmark/benchmark_json/src/BM_Cereal_vs_Simdjson_Test.cpp

```diff
@@ -0,0 +1,525 @@
+/****************************************************************************
+ *
+ *                           Klepsydra Core Modules
+ *              Copyright (C) 2019-2020  Klepsydra Technologies GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *****************************************************************************/
+#define CEREAL_THREAD_SAFE 1
+
+#include <benchmark/benchmark.h>
+#include <iostream>
+#include <simdjson.h>
+#include <vector>
+#include <limits>
+#include <random>
+#include <fstream>  // ofstream
+
+#include <cereal/archives/json.hpp>
+#include <cereal/cereal.hpp>
+#include <cereal/types/vector.hpp>
+
+
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- 
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- 
+bool save_to_myfile(const std::string & str, const std::string & fileName)
+{
+    std::ofstream outFile;
+    outFile.open(fileName);
+    if(!outFile){
+        return false;
+    }
+    outFile << str << std::endl;
+    return true;
+}
+
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- 
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- 
+float get_random_float_using_whole_range(){
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    return std::uniform_real_distribution<float>(std::numeric_limits<float>::max(), std::numeric_limits<float>::min())(gen);
+}
+
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- 
+// cereal needs to know which data members to serialize in your classes. 
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- 
+struct myVectorFloats {
```

Comment:
Use standard naming convention starting with upper case.
The name is not expressive, use `VectorFloatStruct`
Move this class to its own file